packages/render/src/shared/utils/pretty.ts

```diff
@@ -1,101 +1,176 @@
-      };
-    }
+const printProperty = (
+  propertyName: string,
+  propertyValue: string,
+  maxLineLength: number,
+  lineBreak: string,
+) => {
+  const singleLineProperty = `${propertyName}="${propertyValue}"`;
+  if (propertyName === 'style' && singleLineProperty.length > maxLineLength) {
+    // This uses a negative lookbehing to ensure that the semicolon is not
+    // part of an HTML entity (e.g., `&amp;`, `&quot;`, `&nbsp;`, etc.).
+    const nonHtmlEntitySemicolonRegex = /(?<!&[^;]+);/;
+    const styles = propertyValue.split(nonHtmlEntitySemicolonRegex);
+    const wrappedStyles = styles
+      .map((style) => `    ${style}`)
+      .join(`;${lineBreak}`);
+    let multiLineProperty = `${propertyName}="${lineBreak}`;
+    multiLineProperty += `${wrappedStyles}${lineBreak}`;
+    multiLineProperty += `  "`;
 
-    if (doc.type === 'if-break') {
-      return {
-        ...doc,
-        breakContents: recursivelyMapDoc(doc.breakContents, callback),
-        flatContents: recursivelyMapDoc(doc.flatContents, callback),
-      };
-    }
+    return multiLineProperty;
   }
+  return singleLineProperty;
+};
 
-  return callback(doc);
-}
+const printTagStart = (
+  element: HTMLElement,
+  maxLineLength: number,
+  lineBreak: string,
+) => {
+  const singleLineProperties = Object.entries(element.rawAttributes)
+    .map(([name, value]) => ` ${name}="${value}"`)
+    .join('');
+  const singleLineTagStart = `<${element.tagName.toLowerCase()}${singleLineProperties}${element.isVoidElement ? ' /' : ''}>`;
 
-const modifiedHtml = { ...html } as Plugin;
-if (modifiedHtml.printers) {
-  // eslint-disable-next-line @typescript-eslint/unbound-method
-  const previousPrint = modifiedHtml.printers.html.print;
-  modifiedHtml.printers.html.print = (path, options, print, args) => {
-    const node = path.getNode() as HtmlNode;
+  if (singleLineTagStart.length <= maxLineLength) {
+    return singleLineTagStart;
+  }
 
-    const rawPrintingResult = previousPrint(path, options, print, args);
+  let multilineTagStart = `<${element.tagName.toLowerCase()}${lineBreak}`;
+  for (const [name, value] of Object.entries(element.rawAttributes)) {
+    const printedProperty = printProperty(
+      name,
+      value,
+      maxLineLength,
+      lineBreak,
+    );
+    multilineTagStart += `  ${printedProperty}${lineBreak}`;
+  }
+  multilineTagStart += `${element.isVoidElement ? '/' : ''}>`;
+  return multilineTagStart;
+};
 
-    if (node.type === 'ieConditionalComment') {
-      const printingResult = recursivelyMapDoc(rawPrintingResult, (doc) => {
-        if (typeof doc === 'object' && doc.type === 'line') {
-          return doc.soft ? '' : ' ';
-        }
+export const pretty = (html: string, options: Options = {}) => {
+  const root = parse(html, { comment: true });
 
-        return doc;
-      });
+  return printChildrenOf(root, {
+    preserveLinebreaks: false,
+    maxLineLength: 80,
```

Comment:
I think this is an unexpecting behavior for https://github.com/resend/react-email/issues/1847

Although prettier use whitespace sensitive formating https://prettier.io/blog/2018/11/07/1.15.0.html#whitespace-sensitive-formatting 

When we are building email template, the max line is restrictly related to language/fontsize/layout and CSS attributes. I think enforcing an 80 max is a very opinionated decision - the in house formater also carry the same logic that wouldn't fix the issue.

Or any way we can have an option to disable auto line-break for the max length check?